Makefile

```diff
@@ -1,57 +1,54 @@
-.PHONY: all osxkeychain secretservice test lint validate-vendor fmt validate wincred pass deb vendor
+.PHONY: all build-% test lint validate-vendor fmt validate deb vendor
 
-VERSION := $(shell grep 'const Version' credentials/version.go | awk -F'"' '{ print $$2 }')
+PKG := github.com/docker/docker-credential-helpers
+VERSION ?= $(shell git describe --match 'v[0-9]*' --dirty='.m' --always --tags)
+REVISION ?= $(shell git rev-parse HEAD)$(shell if ! git diff --no-ext-diff --quiet --exit-code; then echo .m; fi)
+
+GO_LDFLAGS=-s -w -X ${PKG}/credentials.Version=${VERSION} -X ${PKG}/credentials.Revision=${REVISION} -X ${PKG}/credentials.Package=${PKG}
+
+BUILDX_CMD ?= docker buildx
+DESTDIR ?= ./bin/build
 
 all: test
 
 clean:
 	rm -rf bin
-	rm -rf release
 
-osxkeychain:
-	mkdir -p bin
-	go build -ldflags -s -o bin/docker-credential-osxkeychain osxkeychain/cmd/
+build-%: # build, can be one of build-osxkeychain build-pass build-secretservice build-wincred
+	$(eval BINNAME ?= docker-credential-$*)
+	go build -trimpath -ldflags="$(GO_LDFLAGS) -X ${PKG}/credentials.Name=docker-credential-$*" -o $(DESTDIR)/$(BINNAME) ./$*/cmd/
 
-osxcodesign: osxkeychain
+osxcodesign: build-osxkeychain
 	$(eval SIGNINGHASH = $(shell security find-identity -v -p codesigning | grep "Developer ID Application: Docker Inc" | cut -d ' ' -f 4))
-	xcrun -log codesign -s $(SIGNINGHASH) --force --verbose bin/docker-credential-osxkeychain
-	xcrun codesign --verify --deep --strict --verbose=2 --display bin/docker-credential-osxkeychain
-
-secretservice:
```

Comment:
Can we make the old ones an alias for `build-secretservice` (etc), so that `make secretservice`, `make pass` continues to work?

Reply:
I think technically, these should've been `bin/docker-credential-secretservice`, so that they're matching the artefact that's produced (currently they probably depend on `.PHONY` to prevent them being skipped)

Reply:
Done